flake.nix

```diff
@@ -0,0 +1,105 @@
+{
+  description = "Rust libraries for interacting with hardware from TUXEDO Computers";
+
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable";
+
+    flake-utils.url = "github:numtide/flake-utils";
+
+    fenix = {
+      url = "github:nix-community/fenix";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+
+    pre-commit-hooks = {
+      url = "github:cachix/pre-commit-hooks.nix";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+  };
+
+  outputs = {
+    self,
+    nixpkgs,
+    fenix,
```

Comment:
is there a particular reason why you don't use the nixpkgs default `rustPackages_1_66.rustPlatform` at least if the output is not cached it is much slimmer for me.

Reply:
Mainly because fenix updates the toolchains a bit faster than nixpkgs.
I want to get this working with the fenix first and then I'll see if I can get it working with the nixpkgs version.

You should be able to get the cached toolchain by using the nix-community caches.
If you're on nixos-stable, you might have to [add an overlay](https://github.com/nix-community/fenix/issues/79) in your system config so you can get the cached version in nixos rebuilds.

Reply:
I guess I had the cache as I already use the nix-community cache already, but it downloaded a lot of stuff that seemed to be boilerplate to me, that wasn't necessary with the nixpkgs toolchain. But getting it to work with fenix first sounds reasonable, too. (In particular as the default rustPlatform in nixos 22.11 is older than 1_66 which is necessary to build .)

Reply:
Come to think of it, the fact that fenix uses nixpkgs-stable if you're on nixos-stable might explain why cargo wasn't able to resolve the dependencies in my nixos-rebuild, but was in my nix build (I haven't added the overlay to my system config).